### PR TITLE
Added lparnumascore tests

### PR DESCRIPTION
**Pull request by shirishaganta1**
Tests the lparnumascore command with a list of command line arguments

Signed-off-by: Shirisha Ganta <SHIRISHA.Ganta1@ibm.com>

### CODE REVIEW

**Comment by shirishaganta1** [comment]
on sles
\# avocado run lparnumascore.py -m lparnumascore.py.data/options.yaml 
JOB ID     : e272ed48b722eff729849e2d15b5c341cf09fd9b
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-08-18T11.31-e272ed4/job.log
 (1/1) lparnumascore.py:lparnumascore.test_lparnumascore;run-18a6: CANCEL: lparnumascore is not supported on SuSE (0.22 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-08-18T11.31-e272ed4/results.html
JOB TIME   : 0.51 s

on Rhel8
\# avocado run lparnumascore.py -m lparnumascore.py.data/options.yaml
JOB ID     : fb8929c50cb556b56752830af94899ba5eb4ea62
JOB LOG    : /home/siri/avocado-fvt-wrapper/results/job-2021-08-18T02.29-fb8929c/job.log
 (1/1) lparnumascore.py:lparnumascore.test_lparnumascore;run-18a6: CANCEL:  lparnumascore is not supported on rhel with version 8.Please run on Rhel9 or higher versions only (0.72 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /home/siri/avocado-fvt-wrapper/results/job-2021-08-18T02.29-fb8929c/results.html
JOB TIME   : 15.05 s

on Rhel9
\# avocado run lparnumascore.py -m lparnumascore.py.data/options.yaml 
JOB ID     : 87ed27d129efce6355a27ded5945355ba5a0f797
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-08-18T11.20-87ed27d/job.log
 (1/1) lparnumascore.py:lparnumascore.test_lparnumascore;run-18a6: PASS (0.94 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-08-18T11.20-87ed27d/results.html
JOB TIME   : 1.49 s

File: ras/lparnumascore.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    def run_cmd(self, cmd):
+        if process.system(cmd, ignore_status=True, sudo=True, shell=True):
+            self.fail_cmd.append(cmd)
+        return
+
+    @skipIf(IS_POWER_NV, "This test is supported on PowerVM environment")
+    def setUp(self):
+        det_dist = distro.detect()
+        if det_dist.name == 'SuSE':
```
Comment (PraveenPenguin) [inline ras/lparnumascore.py:44]: just make it simple why not like this 
```
if (det_dist.name == 'rhel' and int(det_dist.version) <= 8) or  det_dist.name == 'SuSE':
  self.cancel("lparnumascore is not supported on %s" %det_dist.name)
```

File: ras/lparnumascore.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                 "Please run on Rhel9 or higher versions only"
+                 % (det_dist.name, det_dist.version))
+        sm = SoftwareManager()
+        if not sm.check_installed("powerpc-utils") and \
+           not sm.install("powerpc-utils"):
+            self.cancel("Fail to install required 'powerpc-utils' package")
+
+    def test_lparnumascore(self):
+        self.log.info("===Executing lparnumascore tool====")
+        self.is_fail = 0
```
Comment (PraveenPenguin) [inline ras/lparnumascore.py:57]: i suggest please make scope this variable to method specific

Comment (shirishaganta1) [reply ras/lparnumascore.py]: Sure. Addressed the above comments and made changes to the script. Please have a look at it once

File: ras/lparnumascore.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            self.cancel("Fail to install required 'powerpc-utils' package")
+
+    def test_lparnumascore(self):
+        self.log.info("===Executing lparnumascore tool====")
+        is_fail = 0
+        self.run_cmd('lparnumascore')
+        lists = self.params.get('list', default=['-c cpu', '-c mem'])
+        for list_item in lists:
+            self.run_cmd('lparnumascore %s' % list_item)
+        if is_fail:
```
Comment (PraveenPenguin) [inline ras/lparnumascore.py:59]: ah seems me we need to have global scope for this variable , my Bad on last review comment sorry , as this code block will never run due to we are not modifying value of `is_fail`

Comment (shirishaganta1) [reply ras/lparnumascore.py]: yes I realised and changed the script

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**